### ISSUE TITLE
set({weekday: 7}) sometimes misses Sunday

### ISSUE THREAD

**Issue by yohcop** [moment/luxon#362]
Hi,

tl;dr

`dateTime.set({weekday: 7})` sometimes (very rarely) selects another day (not Sunday)

```
dt.utc()          ==> .set({weekday: 7})
2016-Jan-02 (Sat) ==> 2016-Jan-02 (Sat)
2016-Dec-29 (Thu) ==> 2017-Jan-02 (Mon)
2021-Jan-01 (Fri) ==> 2021-Jan-04 (Mon)
2028-Jan-01 (Sat) ==> 2028-Jan-01 (Sat)
expected
2016-Jan-02 (Sat) ==> 2016-Jan-03 (Sun)
2016-Dec-29 (Thu) ==> 2017-Jan-01 (Sun)
2021-Jan-01 (Fri) ==> 2021-Jan-03 (Sun)
2028-Jan-01 (Sat) ==> 2028-Jan-02 (Sun)
```
Every other day, beside days in the same week as those examples above, between Jan 2016 and Jan 2028 work as expected. Any day in those weeks will fail too.

The long version:

As I was converting an app to Luxon from Moment, I wanted to make sure I had a few operations correct when they are not available in Luxon. Mostly, startOf/endOf('isoWeek').

I figured this isn't implemented because it's somehow trivial: take your date, set the weekday to Sunday, and then go to the end of day:
`DateTime.local().set({ weekday: 7 }).endOf('day')`

I wrote a big for loop iterating over a thousand days, and compared moment.endOf('isoWeek') and the expression above, to see if I got this right. Mostly, it works. But I found the 4 weeks, whose dates don't work the same as with Moment. I haven't yet figured what's different about those years. For sure they are all the first or last ISO week of the year. I'm sure I can find more examples if needed.

Here's a reproduction: https://stackblitz.com/edit/moment-vs-luxon-endof-isoweek

Thanks,

**Comment by yohcop** [comment]
I got confused about weekDay vs weekday... it's weekday :) definitely.

**Comment by icambron** [comment]
Thanks, definitely looks like a bug. Without debugging it, I suspect it's getting confused by the end of the week year.

**Comment by icambron** [comment]
Fixed in 01546df; I wasn't handling the end of leap years correctly. Released in 1.6.1. Thanks for the report!
